### PR TITLE
GraphQL: Document WishlistItemInterface

### PR DESCRIPTION
**Pull request by keharper**
## Purpose of this pull request

This pull request (PR) adds documentation for the WishlistItemInterface, which was created long ago and extended in https://github.com/magento/magento2/pull/29148.

It also includes updates from MC-37620

~~Note: Another big round of schema changes are coming soon, so I'm not going to update any examples that use the deprecated `Wishlist.item` array.~~

There will not be a separate PR. This PR will contain the final round of schema updates.

## Affected DevDocs pages

-  ...

## Links to Magento source code

-  ...

Added documentation for the GraphQL [WishlistItemInterface](https://devdocs.magento.com/guides/v2.4/graphql/interfaces/wishlist-item-interface.html).

### CODE REVIEW

**Review by rogyar — APPROVED** [review]

File: src/guides/v2.4/graphql/mutations/delete-wishlist.md [hunk 1]
```diff
@@ -23,8 +23,10 @@ The following example deletes a wish list.
 **Request:**
 
 ``` graphql
-mutation {
-  deleteWishlist(wishlistId: 3)
+mutation{
+  deleteWishlist(wishlistId: 5){
+    status
```
Comment (prabhuram93) [inline src/guides/v2.4/graphql/mutations/delete-wishlist.md:28]: Can also highlight the remaining wishlists which will be returned.

**Review by prabhuram93 — CHANGES_REQUESTED** [review]

**Review by erikmarr — APPROVED** [review]
Looks good.

**Review by prabhuram93 — APPROVED** [review]
Looks good @keharper!

**Comment by keharper** [comment]
running tests

**Comment by keharper** [comment]
running tests

**Pull request merged**